tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -87,20 +88,24 @@ func updateMachinePoolQuantity(client *rancher.Client, cluster *v1.SteveAPIObjec
 }
 
 // NewRKEMachinePool is a constructor that sets up a apisV1.RKEMachinePool object to be used to provision a cluster.
-func NewRKEMachinePool(controlPlaneRole, etcdRole, workerRole bool, poolName string, quantity int32, machineConfig *v1.SteveAPIObject, hostnameLengthLimit int, drainBeforeDelete bool) apisV1.RKEMachinePool {
+func NewRKEMachinePool(controlPlaneRole, etcdRole, workerRole bool, poolName string, quantity int32, machineConfig *v1.SteveAPIObject, hostnameLengthLimit int, drainBeforeDelete bool, nodeStartupTimeout *metav1.Duration, unhealthyNodeTimeout *metav1.Duration, maxUnhealthy *string, unhealthyRange *string) apisV1.RKEMachinePool {
```

Comment:
this is getting too long, can you break nodeRoles out in to a similar fashion that provisioninginput->NodePools/Pools follows? And use the new struct as an entry here. I.e.
```

type NodeRoles struct {
	ControlPlane         bool             `json:"controlplane,omitempty" yaml:"controlplane,omitempty"`
	Etcd                 bool             `json:"etcd,omitempty" yaml:"etcd,omitempty"`
	Worker               bool             `json:"worker,omitempty" yaml:"worker,omitempty"`
	Windows              bool             `json:"windows,omitempty" yaml:"windows,omitempty"`
	Quantity             int32            `json:"quantity" yaml:"quantity"`
}

type MachineNode struct {
        NodeRoles
	DrainBeforeDelete    bool             `json:"drainBeforeDelete" yaml:"drainBeforeDelete"`
	NodeStartupTimeout   *metav1.Duration `json:"nodeStartupTimeout,omitempty" yaml:"nodeStartupTimeout,omitempty"`
	UnhealthyNodeTimeout *metav1.Duration `json:"unhealthyNodeTimeout,omitempty" yaml:"unhealthyNodeTimeout,omitempty"`
	MaxUnhealthy         *string          `json:"maxUnhealthy,omitempty" yaml:"maxUnhealthy,omitempty"`
	UnhealthyRange       *string          `json:"unhealthyRange,omitempty" yaml:"unhealthyRange,omitempty"`
}

```